gensim/corpora/textcorpus.py

```diff
@@ -191,19 +203,34 @@ def __init__(self, source=None, dictionary=None, metadata=False, character_filte
                 will be built for the given corpus. If no corpus is given, the dictionary will
                 remain uninitialized.
             metadata (bool): True to yield metadata with each document, else False (default).
+                By default, metadata consists only of the line number for each document. This
+                is yielded from `get_texts` as a tuple of (tokens, (line_num,)) and from __iter__
+                as a tuple of (bag_of_words, (line_num,)).
             character_filters (iterable of callable): each will be applied to the text of each
                 document in order, and should return a single string with the modified text.
-                For Python 2, the original text will not be unicode, so it may be useful to
-                convert to unicode as the first character filter. The default character filters
-                lowercase, convert to unicode (strict utf8), perform ASCII-folding, then collapse
+                For Python 2, the original text will not be unicode (unless you modify your
```

Comment:
I propose dropping the `(unless ...` bracket. This is already very complicated as it is.

Also, why single out Python 2? Is the behaviour different between Python 2 vs Python 3?
If so, I'd consider that a bug.

Let's keep the API as simple as possible: `getstream` returns unicode (no matter the Python version); all filters expect unicode.